docs/troubleshooting/Performance.mdx

```diff
@@ -79,7 +79,113 @@ module.exports = {
 
 See [Glob pattern in parser's option "project" slows down linting](https://github.com/typescript-eslint/typescript-eslint/issues/2611) for more details.
 
-## The `indent` / `@typescript-eslint/indent` rules
+## Changes to `extraFileExtensions` with `projectService` in your ESLint options
+
+Using different [`extraFileExtensions`](../packages/Parser.mdx#extrafileextensions) for different files with
+[`projectService`](../packages/Parser.mdx#projectservice) causes the TypeScript server to perform a full project reload.
+Project reloads can be observed using the [debug environment variable](../packages/typescript-estree/#debugging): `DEBUG='typescript-eslint:typescript-estree:*'`.
+
+```
+typescript-estree:useProgramFromProjectService Updating extra file extensions: before=[]: after=[ '.vue' ]
```

Comment:
I trimmed these log messages on the left so they would (mostly) fit horizontally on screen.  If accuracy is preferred, I can undo this.

```typescript-estree:useProgramFromProjectService```

vs

```2024-06-17T16:58:28.942Z typescript-eslint:typescript-estree:useProgramFromProjectService```

Reply:
Yeah +1 this is preferable IMO. I don't think users will be upset at not having to read through arbitrary timestamps. 😄

Reply:
That being said, could you move this debug information to the bottom? I think it'd be easier for folks to have the order be:

1. Introduction to this idea (don't change `extraFileExtensions` between files)
2. Why this is the case (full project reloads)
3. Ancillary info, like debugging tips